crypto/rsa/rsa_pss.c

```diff
@@ -110,7 +111,9 @@ int RSA_verify_PKCS1_PSS_mgf1(RSA *rsa, const unsigned char *mHash,
         ERR_raise(ERR_LIB_RSA, RSA_R_SLEN_RECOVERY_FAILED);
         goto err;
     }
-    if (sLen != RSA_PSS_SALTLEN_AUTO && (maskedDBLen - i) != sLen) {
+    if (sLen != RSA_PSS_SALTLEN_AUTO
+            && sLen != RSA_PSS_SALTLEN_AUTO_DIGEST_MAX
```

Comment:
You could actually compute the expected sLen with RSA_PSS_SALTLEN_AUTO_DIGEST_MAX so IMO it is not necessary to autorecover it.

Reply:
That depends on the behavior you want. For maximum compatibility, I want the default for verification to behave exactly as it did before.

If I change this to expect the specific value that RSA_PSS_SALTLEN_AUTO_DIGEST_MAX would create when signing, there's a chance that a number of signatures that have so far verified without problem will now suddenly fail to verify.

Or to put it in a different way: I don't want to fail because the expected salt length is not what this option would have produced. If we want to limit what the FIPS provider will accept, we should do that in the FIPS provider, and only there.

Reply:
OK, that makes sense.